contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,146 @@
+//SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.7.3;
+
+import {IVolatilityOracle} from "../interfaces/IVolatilityOracle.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+import {ICompToken} from "../interfaces/ICompToken.sol";
+import {DSMath} from "../libraries/DSMath.sol";
+import {IBlackScholes} from "../interfaces/IBlackScholes.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {Math} from "../libraries/Math.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import "hardhat/console.sol";
+
+contract OptionsPremiumPricer is DSMath {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IVolatilityOracle public immutable volatilityOracle;
+    IPriceOracle public immutable priceOracle;
+    ICompToken public immutable cToken;
+    // IKEEP3rVolatility: 0xCCdfCB72753CfD55C5afF5d98eA5f9C43be9659d
+    IBlackScholes public immutable blackScholes;
+
+    constructor(
+        address _volatilityOracle,
+        address _priceOracle,
+        address _cToken,
+        address _blackScholes
+    ) {
+        require(_volatilityOracle != address(0), "!_volatilityOracle");
+        require(_priceOracle != address(0), "!_priceOracle");
+        require(_cToken != address(0), "!_cToken");
+        require(_blackScholes != address(0), "!_blackScholes");
+        volatilityOracle = IVolatilityOracle(_volatilityOracle);
+        priceOracle = IPriceOracle(_priceOracle);
+        cToken = ICompToken(_cToken);
+        blackScholes = IBlackScholes(_blackScholes);
+    }
+
+    /**
+     * @notice Calculates the premium of the provided option using Black-Scholes
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @param isPut is whether the option is a put option
+     */
+    function getPremium(
+        uint256 st,
+        uint256 expiryTimestamp,
+        bool isPut
+    ) external view returns (uint256 premium) {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        uint256 sp = priceOracle.latestAnswer().div(10**priceOracle.decimals());
+        uint256 v = volatilityOracle.vol();
```

Comment:
blackScholes.quotePrice(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2, 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48, t, sp, st);

If we want to use vol from IKeeprV3 instead of homemade vol